mewarpx/mewarpx/emission.py

```diff
@@ -1461,64 +1559,90 @@ class VolumeEmitter(BaseEmitter):
     """Parent class for volumetric particle injection coordinates.
 
         - ``volume`` gives the spatial volume in m^3
-        - ``_get_xv_coords()`` implements the subclass-specific particle
+        - ``_get_x_coords()`` implements the subclass-specific particle
           injection logic
     """
 
     volume = 0
+    geoms = ['Z', 'XZ', 'RZ', 'XYZ']
 
-    def __init__(self, T):
-        """Default initialization for all Emitter objects.
+    def __init__(self, T, xmin=None, xmax=None, ymin=None, ymax=None,
+                 zmin=None, zmax=None, rmin=None, rmax=None):
+        """Initialize emitter boundaries. A rectangular or cylindrical emitter
+        volume is supported. If x & y boundaries are specified the r boundaries
+        will be ignored and vice versa. If both x & y and r boundaries are
+        specified an AttributeError will be raised. If no boundaries are given
+        the simulation geometry and boundaries will be used.
 
         Arguments:
             T (float): Emitter temperature in Kelvin. Determines particle
                 velocity distribution.
+            x/y/z/rmin (float): Lower boundary of the volume.
+            x/y/z/rmax (float): Upper boundary of the volume.
         """
         super(VolumeEmitter, self).__init__()
         self.T = T
 
-    def getvoltage(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return 0.
-
-    def getvoltage_e(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return self.getvoltage()
-
+        # determine default prism type from simulation geometry
+        self.rectangular = mwxrun.geom_str != 'RZ'
 
-class CartesianVolumeEmitter(VolumeEmitter):
-
-    """VolumeEmitter with bounds specified by x/y/z rectangular prism.
+        # check if a different volume was specified
+        r_bounds_given = (rmin is not None or rmax is not None)
+        xy_bounds_given = (
+            xmin is not None or xmax is not None or ymin is not None or
+            ymax is not None
+        )
 
-    A template ``_get_xv_coords()`` is also provided but can be overridden if
-    needed. It calls ``_get_x_coords()``.
-    """
+        if r_bounds_given and xy_bounds_given:
+            raise AttributeError(
+                "Both rectangular and cylindrical boundaries specified for a "
+                "VolumeEmitter"
+            )
```

Comment:
Also add a check if neither are given.

Reply:
If neither are given the simulation domain will be used along with the simulation geometry. This also addresses the overall comment you made (defaults will be properly set using the simulation properties) and the comment above (the `rectangular` flag will only be overwritten if either rectangular or cylindrical boundaries are specified).

Reply:
I do agree that the cylindrical prism won't really be useful until we consider 3d simulations but I don't see a reason not to build the code as general as possible from the start.